### ISSUE TITLE
Possibly unhandled Error: 1 at Promise$_rejecter 

### ISSUE THREAD

**Issue by crapthings** [jkrup/meteor-now#50]
what's wrong ?
should i use same version with meteor node -v ?

should i always use mongodb ? because it's just a client bundle.

```

WARNING: Did not pass a MONGO_URL. Bundling a NON-PRODUCTION version of MongoDB with your application. Read about the limitations here: https://git.io/vM72E
⠀⢀ [METEOR-NOW] - deploying build (this can take several minutes)Possibly unhandled Error: 1
    at Promise$_rejecter (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:590:44)
    at /Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/index.js:85:13
    at tryCatch1 (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/util.js:63:19)
    at Promise$_callHandler [as _callHandler] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:695:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:711:18)
    at Promise$_settlePromiseAt [as _settlePromiseAt] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:868:14)
    at Promise$_settlePromises [as _settlePromises] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:1006:14)
    at Promise$_rejectPromises [as _rejectPromises] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:999:10)
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/async.js:74:12)
    at Async$consumeFunctionBuffer (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/async.js:37:14)
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)

```

then [METEOR-NOW] - deploying build (this can take several minutes) forever

**Comment by crapthings** [comment]
i have set mongourl and i've still got

```
⡋⠁ [METEOR-NOW] - deploying build (this can take several minutes)Possibly unhandled Error: 1
    at Promise$_rejecter (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:590:44)
    at /Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/index.js:85:13
    at tryCatch1 (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/util.js:63:19)
    at Promise$_callHandler [as _callHandler] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:695:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:711:18)
    at Promise$_settlePromiseAt [as _settlePromiseAt] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:868:14)
    at Promise$_settlePromises [as _settlePromises] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:1006:14)
    at Promise$_rejectPromises [as _rejectPromises] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/promise.js:999:10)
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/async.js:74:12)
    at Async$consumeFunctionBuffer (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/meteor-now/node_modules/promise-spawner/node_modules/bluebird/js/main/async.js:37:14)
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

**Comment by crapthings** [comment]
-d logs

```
> [debug] Retrying: FetchError: request to https://api.zeit.co/now/sync failed, reason: socket hang up
FetchError: request to https://api.zeit.co/now/sync failed, reason: socket hang up
    at ClientRequest.<anonymous> (/Users/monsterstep/.nvm/versions/node/v6.10.2/lib/node_modules/now/node_modules/node-fetch/index.js:133:11)
    at emitOne (events.js:101:20)
    at ClientRequest.emit (events.js:188:7)
    at TLSSocket.socketOnEnd (_http_client.js:345:9)
    at emitNone (events.js:91:20)
    at TLSSocket.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

**Comment by acomito** [comment]
did you make sure to npm install for your meteor prooject? Maybe try to run meteor as well so that it downloads all the atmosphere packages.

**Comment by purplecones** [comment]
Might be related to the now-cli. According to the last log you posted, the error is related with the socket connection so maybe you lost connection during upload or some error occurred on now's end. I good simple test is to just deploy a super basic node app and see if that deploys successfully by just typing in `now` in your project folder.

**Comment by UShan89** [comment]
I recently solved this issue. 

#47 has more info. Make your `meteor-now -e 'MONGO_URL=mongodb://<>'  `

i.e. single quote the whole as a string

**Comment by crapthings** [comment]
hey it still throw the error, but it did deploy the project.

and there's no node-fetch used in this project, the project is very small, its almost static html with small amount js basiclly

**Comment by UShan89** [comment]
Please try this 

`meteor npm install && meteor reset && meteor-now`  

it takes time because the Docker image created for deployment installs a non-persistent mongodb on now. 

If you use your own mongo_url and point to a server accessible over internet, the build is much faster.

**Comment by qualitymanifest** [comment]
I've had 4 successful deployments, the latest one being yesterday, but today I keep getting this error. I've tried the following:
- Single quoting `'MONGO_URL=mlab uri...'`
- Since that didn't work, also single quoting `'ROOT_URL=...'`
- Rolling back to now@4.11.1 since I just upgraded to now@4.11.2.
- `rm -rf node_modules` followed by `meteor npm install` and `meteor reset`.
Nothing has worked, I keep getting the error and then it hangs at `deploying build`. I'm running Meteor@1.4.4.1

**Comment by crapthings** [comment]
@qualitymanifest 

meteor now -d -e ...rest

error shows, but it did deploy successed

**Comment by UShan89** [comment]
@crapthings I had the exact same error, Please use single quotes for the whole of your MONGO_URL options. 

Instance: `meteor-now -e 'MONGO_URL=mongodb://my_mango_server:27017/the_db?ssl=true' ` 

The only other thing I can think of is that your DB is not internet connected OR firewall rule is not allowing incoming port connection

**Comment by qualitymanifest** [comment]
@crapthings I could see it was deploying before, by looking on my dashboard on zeit's website. But it wasn't completing the deployment - it would get stuck on "syncing files".
Anyway, I tried your suggestion of passing in -d with the rest of the command that I was using. I didn't get the error at all and it deployed just fine. I also logged out and logged back in first, so it's possible that had something to do with it, since I hadn't tried that before. Who knows at this point, I'm just happy it's working again so I don't keep burning through my free deployments. Thanks!

**Comment by d62remi** [comment]
To fix this issue you can try 
`npm install now@4.10.3 -g`

I think problem is cause by https://github.com/zeit/now-cli/pull/435 **only for oss plan**. 

Now-cli is waiting for your answer to go on https://github.com/zeit/now-cli/pull/435/files#diff-f784a12f4f3daf2399f86ae70b2f69c5R617 

@jkrup @purplecones an idea to fix this problem with last now-cli ?

**Comment by nilsi** [comment]
I have this problem as well I  did `npm install now@4.10.3 -g` but it did now help.

**Comment by jkrup** [comment]
What version of `node` do you have? (`node --version`)

**Comment by nilsi** [comment]
```
$ node --version
v5.8.0
```

**Comment by fknipp** [comment]
I ran into similar problems and solved them by performing two steps:

- Installing node version 6.9.0
- Installing now version 4.10.3

```
sudo npm install -g n
sudo n 6.9.0
sudo npm install -g now@4.10.3
```

I've run the **meteor-now** command always with **-d**, I didn't try, if it works without the switch as well.

**Comment by webmagnets** [comment]
I was getting this error and using -d worked for me. For some reason the prompt "Are you sure you want to proceed with the deployment? [y|N] " only comes up if you use "-d".

**Comment by jkrup** [comment]
These issues should hopefully be fixed after our re-write #75 so I'm closing this thread. Please open another issue if you have any troubles with the latest version.
